pkg/cloud/services/elb/eip.go

```diff
@@ -0,0 +1,50 @@
+package elb
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/elbv2"
+
+	infrav1 "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
+)
+
+func getElasticIPRoleName() string {
+	return fmt.Sprintf("lb-%s", infrav1.APIServerRoleTagValue)
+}
+
+func (s *Service) allocatedPublicIpv4AddressFromByoIPPool(input *elbv2.CreateLoadBalancerInput) error {
+	// Custom Public IPv4 Pool isn't set.
+	if s.scope.VPC().GetPublicIpv4Pool() == nil {
+		return nil
+	}
+
+	// Only NLB is supported
+	if *input.Type != string(elbv2.LoadBalancerTypeEnumNetwork) {
+		return fmt.Errorf("custom PublicIpv4Pool is supported only with Network Load Balancer type: %s", *input.Type)
+	}
+
+	// Custom SubnetMappings should not be defined or overridden by user-defined mapping.
+	if len(input.SubnetMappings) > 0 {
+		return fmt.Errorf("custom PublicIpv4Pool is mutual exclusive with SubnetMappings: %v", input.SubnetMappings)
```

Comment:
Nit: I don't know if there is any benefit to adding the subnet mappings in the error message. The issue is not with those mappings themselves but with any subnet mappings being specified.
```suggestion
		return fmt.Errorf("custom PublicIpv4Pool is mutually exclusive with SubnetMappings")
```